instruqt/closing-information-gap/04-adding-to-the-chart/assignment.md

```diff
@@ -0,0 +1,143 @@
+---
+slug: adding-to-the-chart
+id: tvkvu4hgnpuv
+type: challenge
+title: Adding the Support Bundle to the Harbor Helm Chart
+teaser: Learn how to incorporate your support bundle into your chart
+notes:
+- type: text
+  contents: Time to add your support bundle definition to the Harbor Helm chart
+tabs:
+- title: Shell
+  type: terminal
+  hostname: shell
+- title: Manifest Editor
+  type: code
+  hostname: shell
+  path: /home/replicant
+difficulty: basic
+timelimit: 600
+---
+
+You've put together a pretty robust support bundle definition for the Harbor
+registry. Now it's time to distribute that definition to your customers so that
+they can collect a support bundle when you need it.
+
+Making the Support Bundle Available
+===================================
+
+The Support Bundle specification looks a lot like Kubernetes customer resource,
+but strictly speaking it isn't. Instead it runs outside of the cluster and is
+interpreted by the `support-bundle` plugin to the `kubectl` command.
+
+Even if it isn't a Kubernetes resources, we still want to distribute the
+support bundle specification as part of the application Helm chart. This saves
+you from having to find a different distribution mechanism. It also makes it
+easier for your customer to gather a support bundle for you.
+
+To include the support bundle definition with your application, you're going to
+store it in a Kubernetes secret. There are a couple of constraints on this
+secret:
+
+1. It has to have a label `troubleshoot.sh/kind: support-bundle`
+2. You need to use the key `support-bundle-spec` for your support bundle
+   specification.
+
+If we consider the baseline support bundle from the first step in the lab, it
+becomes a secret that looks like this
+
+```
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: simplest-support-bundle
+  labels:
+    troubleshoot.sh/kind: support-bundle
+stringData:
+  support-bundle-spec: |-
+    apiVersion: troubleshoot.sh/v1beta2
+    kind: SupportBundle
+    metadata:
+      name: simplest-support-bundle
+    spec:
+      collectors: []
+      analyzers: []
+```
+
+Let's add this secret to our cluster and execute it.
+
+```
+kubectl create -f simplest-support-bundle-secret.yaml
```

Comment:
the secret here doesn't have the required label, so kubectl doesn't find it when we install it